### ISSUE TITLE
Fix overridden method signature: logits_to_keep -> output_router_logits in AfmoeForCausalLM at modular_afmoe.py

### ISSUE THREAD

**Issue by deborshi-web** [huggingface/transformers#48506]
Parameter 'logits_to_keep' has been renamed to 'output_router_logits' in overriding 'AfmoeForCausalLM.forward' method (arguments-renamed)

**Comment by vasqu** [comment]
I think you are confused? 
- logits to keep is meant for the lm head calculation (i.e. how many logits of the sequence do you keep) which is useful for generate related things where you only need the last logit for example
- output router logits controls whether you output the MoE router logits

2 different things so closing this unless you have something else

**Comment by deborshi-web** [comment]
> I think you are confused? 
> - logits to keep is meant for the lm head calculation (i.e. how many logits of the sequence do you keep) which is useful for generate related things where you only need the last logit for example
> - output router logits controls whether you output the MoE router logits
> 
> 2 different things so closing this unless you have something else

I just want to ask a question Method 'get_position_embeddings' is abstract in class 'PreTrainedModel' but is not overridden in child class 'AfmoeForCausalLM' (abstract-method) is it normal

**Comment by vasqu** [comment]
Yep, pretty normal. It usually refers to something like the sin/cos positional embeddings of a few models. But it's also very old and not really used by us or others
